### PR TITLE
fix: [CDS-72049]: Replace K8s instance sync validation task to PT validation task

### PR DESCRIPTION
**Pull request by abosii-harness**
## Describe your changes
Current task is doing full fetch of the instance sync instead of just confirming if delegate is capable of using delegate or not. This results in PTs not being executed and the status is not reported to the instance handler. 

Changing this validation task to just checking delegate capability 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49549)

### CODE REVIEW

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks

**Comment by abosii-harness** [comment]
trigger uts

File: 400-rest/src/main/java/io/harness/perpetualtask/instancesync/ContainerInstanceSyncPerpetualTaskClient.java [hunk 1]
```diff
@@ -170,15 +170,17 @@ private DelegateTask buildK8sDelegateTask(
                                                            .releaseName(taskData.getReleaseName())
                                                            .build();
 
+    List<ExecutionCapability> executionCapabilities = delegateTaskParams.fetchRequiredExecutionCapabilities(null);
+
     return DelegateTask.builder()
```
Comment (acasian-harness) [inline 400-rest/src/main/java/io/harness/perpetualtask/instancesync/ContainerInstanceSyncPerpetualTaskClient.java:175]: isn't this part common for both the methods?

**Review by acasian-harness — APPROVED** [review]

File: 400-rest/src/main/java/io/harness/perpetualtask/instancesync/ContainerInstanceSyncPerpetualTaskClient.java (hunk 1 above)
Comment (abosii-harness) [reply 400-rest/src/main/java/io/harness/perpetualtask/instancesync/ContainerInstanceSyncPerpetualTaskClient.java:175]: If you mean buildNonK8sDelegateTask, then no. There are too many differences to make this common, i.e. tags, executionCapabiltiies, task type and task parameters. Not worth to make it common

**Review by GoelPratyush99 — APPROVED** [review]

**Review by alebuh — APPROVED** [review]

**Review by acasian-harness — APPROVED** [review]

**Comment by abosii-harness** [comment]
trigger review, messagecheck

**Pull request merged**